packages-exp/auth-exp/src/core/util/delay.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isMobileCordova, isReactNative } from '@firebase/util';
+import { isOnline } from './navigator';
+
+export const _OFFLINE_DELAY_MS = 5000;
+
+/**
+ * A structure to help pick between a range of long and short delay durations
+ * depending on the current environment. In general, the long delay is used for
+ * mobile environments whereas short delays are used for desktop environments.
+ */
+export class Delay {
+  // The default value for the offline delay timeout in ms.
+
+  private readonly isMobile: boolean;
+  constructor(
+    private readonly shortDelay: number,
+    private readonly longDelay: number
+  ) {
+    // Internal error when improperly initialized.
+    if (shortDelay > longDelay) {
+      throw new Error('Short delay should be less than long delay!');
```

Comment:
I suppose this is a debug assertion that can be removed in prod build?

Reply:
Not that I am looking at this code, but since you are adding a new code base: You should follow the model we added in Firstore and have "debug asserts" and "hard asserts". The debug asserts can then be stripped from the production build, which saves a lot of bytes.

Reply:
updated, ready for re-review